wodles/gcloud/pubsub/subscriber.py

```diff
@@ -0,0 +1,158 @@
+            Project name
+        subscription_id : str
+            Subscription ID
+
+        Returns
+        -------
+        String with the subscription path
+        """
+        return self.subscriber.subscription_path(project, subscription_id)
+
+    def check_permissions(self):
+        """Check if permissions are OK for executing the wodle."""
+        required_permissions = {'pubsub.subscriptions.consume'}
+        response = self.subscriber.test_iam_permissions(request={'resource': self.subscription_path,
+                                                                 'permissions': required_permissions})
+        if required_permissions.difference(response.permissions) != set():
+            error_message = 'ERROR: No permissions for executing the wodle from this subscription'
+            raise Exception(error_message)
+
+    def format_msg(self, msg: bytes) -> str:
+        """Format a message.
+        :param msg: Message to be formatted
+        """
+        # Insert msg as value of self.key_name key.
+        return f'{{"integration": "gcp", "{self.key_name}": {msg.decode(errors="replace")}}}'
+
+    def send_message(self, message):
+        """Send a message with a header to the analysisd queue.
+        :param message: Message to send to the analysisd queue
+        """
+        self.wazuh_queue.send(f'{self.header}{message}'.encode(errors='replace'))
+
+    def pull_request(self, max_messages) -> int:
+        """Make request for pulling messages from the subscription and acknowledge them.
+        :param max_messages: Maximum number of messages to retrieve
+        :return: Number of processed messages
+        """
+        try:
+            response = self.subscriber.pull(
+                request={'subscription': self.subscription_path,
+                         'max_messages': max_messages}
+            )
+        except google.api_core.exceptions.DeadlineExceeded:
+            self.logger.warning('Deadline exceeded when pulling messages. No more messages will be retrieved on this '
+                                'execution')
+            return 0
+
+        ack_ids = []
+        for received_message in response.received_messages:
+            formatted_message = self.format_msg(received_message.message.data)
+            self.logger.debug(f'Processing event: {formatted_message}')
+            ack_ids.append(received_message.ack_id)
+            self.send_message(formatted_message)
+
+        ack_ids and self.subscriber.acknowledge(
+            request={'subscription': self.subscription_path, 'ack_ids': ack_ids}
+        )
+
+        return len(response.received_messages)
+
+    def process_messages(self, max_messages: int = 100) -> int:
+        """Process the available messages in the subscription.
+        :param max_messages: Maximum number of messages to retrieve
+        :return: Number of processed messages
+        """
+        try:
+            self.wazuh_queue.connect(utils.ANALYSISD)
```

Comment:
Create a new function called `connect` in `WazuhGCloudIntegration` to handle the socket connection instead